include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -61,39 +61,34 @@ class bi_fm_index_cursor;
  * cannot occur in the text.
  *
  * This index will only work for byte alphabets, i.e. alphabets with a size <= 256.
- * When switching to bigger alphabets, the requires clause of the seqan3::fm_index, the seqan3::fm_index_default_traits,
- * the delimiter choice and the transform view for the construction need to be adjusted. Additionally, all occurrences
- * of uint8_t should be double checked to make sure they also apply to bigger alphabets.
+ * When switching to bigger alphabets, the static_asserts, the delimiter choice and the transform view for the
+ * construction need to be adjusted. Additionally, all occurrences of uint8_t should be double checked to make sure
+ * they also apply to bigger alphabets.
  * \endif
  *
  * \f$T_{BACKWARD\_SEARCH}: O(\log \Sigma)\f$
  *
  * \todo Asymptotic space consumption:
  *
  */
-struct fm_index_default_traits
-{
-    //!\brief Type of the underlying SDSL index.
-    using sdsl_index_type = sdsl::csa_wt<
-        sdsl::wt_blcd<
-            sdsl::bit_vector,
-            sdsl::rank_support_v<>,
-            sdsl::select_support_scan<>,
-            sdsl::select_support_scan<0>
-        >,
-        16,
-        10000000,
-        sdsl::sa_order_sa_sampling<>,
-        sdsl::isa_sampling<>,
-        sdsl::plain_byte_alphabet
-    >;
-};
+using sdsl_index_default_type = sdsl::csa_wt<
+                                             sdsl::wt_blcd<
+                                                 sdsl::bit_vector,
+                                                 sdsl::rank_support_v<>,
+                                                 sdsl::select_support_scan<>,
+                                                 sdsl::select_support_scan<0>
+                                             >,
+                                             16,
+                                             10000000,
+                                             sdsl::sa_order_sa_sampling<>,
+                                             sdsl::isa_sampling<>,
+                                             sdsl::plain_byte_alphabet
+                                >;
```

Comment:
```
using sdsl_index_default_type = 
    sdsl::csa_wt<sdsl::wt_blcd<sdsl::bit_vector,
                               sdsl::rank_support_v<>,
                               sdsl::select_support_scan<>,
                               sdsl::select_support_scan<0>>,
                 16,
                 10000000,
                 sdsl::sa_order_sa_sampling<>,
                 sdsl::isa_sampling<>,
                 sdsl::plain_byte_alphabet>;
```

Reply:
I would maybe directly rename it to  `sdsl_wt_index_type`. And then define `using default_sdsl_index_type = sdsl_wt_index_type;`... ?

Reply:
Then people can explicitly use `wt` already and when we add `sdsl_epr_index_type` we can change the default. Also I would move the `default` keyword to front...